### PR TITLE
Migrate usage of SSM to aws-sdk-go-v2

### LINKED ISSUE #4880: migrate to aws-sdk-go-v2 [4/9]
## Background

As part of #3215 we want to migrate to use the [aws-sdk-go v2](https://github.com/aws/aws-sdk-go-v2). This involves migrating all of the AWS API calls, roughly 500~ to use the new package. A spike was complete and its finding was documented [here](https://github.com/weaveworks/eksctl/issues/3215#issuecomment-1047099306) and that included a demo branch [here](https://github.com/weaveworks/eksctl/pull/4820/files). As the spike documents we can split out the work to migrate to v2 to be done in 9 parts, were we essentially use both v1 and v2 of the API while we migrate.

This ticket is to: remove usage of `v1` api for `SSM` and replace it with `v2` 

Blocked on #4877. Can be done in parallel to parts 2-8

Also, when migrating to a new call we will propagate the context from the command down to the AWS calls.

### PR DESCRIPTION
**Pull request by cPu1**
### Description

Migrates usage of SSM to aws-sdk-go-v2.

Propagates the context to all calls to the SSM API and for the `ec2.DescribeImages` call. The remaining context changes for EC2 and EKS in the changed codepaths will be updated in #4883 and #4884. `context.TODO` is used as the top-level context to ensure we use an appropriate context later, which is outside of the scope of this PR.

Closes #4880

### Checklist
- [ ] Added tests that cover your change (if possible)
- [ ] Added/modified documentation as required (such as the `README.md`, or the `userdocs` directory)
- [ ] Manually tested
- [x] Made sure the title of the PR is a good description that can go into the release notes
- [x] (Core team) Added labels for change area (e.g. `area/nodegroup`) and kind (e.g. `kind/improvement`)

### BONUS POINTS checklist: complete for good vibes and maybe prizes?! :exploding_head:
- [ ] Backfilled missing tests for code in same general area :tada:
- [ ] Refactored something and made the world a better place :star2:

### CODE REVIEW

**Review by Skarlso — APPROVED** [review]
LGTM! Nice :)

File: pkg/actions/nodegroup/create.go [hunk 1]
```diff
@@ -80,7 +81,7 @@ func (m *Manager) Create(options CreateOpts, nodegroupFilter filter.NodegroupFil
 	}
 
 	if !options.DryRun {
-		if err := m.init.Normalize(nodePools, cfg.Metadata); err != nil {
+		if err := m.init.Normalize(context.TODO(), nodePools, cfg.Metadata); err != nil {
```
Comment (aclevername) [inline pkg/actions/nodegroup/create.go:84]:
```suggestion
		if err := m.init.Normalize(context.TODO(), nodePools, cfg.Metadata); err != nil {
```

can we pass the ctx down from `ctl/nodegroup/create.go`?

**Review by aclevername — APPROVED** [review]
LGTM, just the 1 place where we aren't correctly passing the context down :smile:

File: pkg/actions/nodegroup/create.go (hunk 1 above)
Comment (cPu1) [reply pkg/actions/nodegroup/create.go]: Done.

**Pull request merged**